lib/icinga/icingaapplication.cpp

```diff
@@ -289,6 +292,40 @@ String IcingaApplication::GetNodeName() const
 	return ScriptGlobal::Get("NodeName");
 }
 
+/*
+ * Setting up PluginDir default value as a ScriptGlobal
+ *
+ * Note: only PluginDir is used inside the C++ code directly
+ * All other globals are set for compatibility, and a reduced constants.conf
+ */
+void IcingaApplication::DeclarePluginDir(const String& pluginDir)
+{
+	for (String constant : {"PluginDir", "ManubulonPluginDir", "PluginContribDir"})
+		if (!ScriptGlobal::Exists(constant))
+			ScriptGlobal::Set(constant, pluginDir);
+}
+
+String IcingaApplication::GetPluginDir(void)
+{
+	return ScriptGlobal::Get("PluginDir");
+}
+
+void IcingaApplication::DeclarePluginPath(const String& pluginPath)
+{
+	if (!ScriptGlobal::Exists("PluginPath"))
+		ScriptGlobal::Set("PluginPath", Array::FromVector(String(pluginPath).Split(":")));
+}
+
+Value IcingaApplication::GetPluginPath(void)
```

Comment:
This method should really just return an `Array::Ptr` because you're already coercing the return value of `ScriptGlobal::Get` into an `Array::Ptr`.